Maple2.Server.Game/PacketHandlers/ClubHandler.cs

```diff
@@ -0,0 +1,230 @@
+﻿using Grpc.Core;
+using Maple2.Database.Storage;
+using Maple2.Model.Error;
+using Maple2.Model.Game.Club;
+using Maple2.Model.Game.Party;
+using Maple2.Model.Metadata;
+using Maple2.PacketLib.Tools;
+using Maple2.Server.Core.Constants;
+using Maple2.Server.Core.PacketHandlers;
+using Maple2.Server.Game.Manager;
+using Maple2.Server.Game.Packets;
+using Maple2.Server.Game.Session;
+using Maple2.Server.World.Service;
+using WorldClient = Maple2.Server.World.Service.World.WorldClient;
+using ClubResponse = Maple2.Server.World.Service.ClubResponse;
+
+namespace Maple2.Server.Game.PacketHandlers;
+
+public class ClubHandler : PacketHandler<GameSession> {
+    public override RecvOp OpCode => RecvOp.Club;
+
+    private enum Command : byte {
+        Create = 1,
+        StagedInvite = 3,
+        Invite = 6,
+        RespondInvite = 8,
+        Leave = 10,
+        Buff = 13,
+        Rename = 14,
+    }
+
+    #region Autofac Autowired
+    // ReSharper disable MemberCanBePrivate.Global
+    public required WorldClient World { private get; init; }
+    // ReSharper restore All
+    #endregion
+
+    public override void Handle(GameSession session, IByteReader packet) {
+        var command = packet.Read<Command>();
+        switch (command) {
+            case Command.Create:
+                HandleCreate(session, packet);
+                break;
+            case Command.StagedInvite:
+                HandleStagedInvite(session, packet);
+                break;
+            case Command.Invite:
+                HandleInvite(session, packet);
+                break;
+            case Command.RespondInvite:
+                HandleRespondInvite(session, packet);
+                break;
+            case Command.Leave:
+                HandleLeave(session, packet);
+                break;
+            // case Command.Buff:
+            //     Buff(session, packet);
+            //     break;
+            case Command.Rename:
+                HandleRename(session, packet);
+                break;
+        }
+    }
+
+    private void HandleCreate(GameSession session, IByteReader packet) {
+        string clubName = packet.ReadUnicodeString();
+
+        Party? party = session.Party.Party;
```

Comment:
`party`??

Reply:
Clubs can only be created via a party leader.

Reply:
got it, can you add a comment it looked like a copy-paste bug to me